src/runtime/components/forms/Form.vue

```diff
@@ -81,18 +87,12 @@ export default defineComponent({
       let errs = await props.validate(props.state)
 
       if (props.schema) {
-        if (isZodSchema(props.schema)) {
-          errs = errs.concat(await getZodErrors(props.state, props.schema))
-        } else if (isYupSchema(props.schema)) {
-          errs = errs.concat(await getYupErrors(props.state, props.schema))
-        } else if (isJoiSchema(props.schema)) {
-          errs = errs.concat(await getJoiErrors(props.state, props.schema))
-        } else if (isValibotSchema(props.schema)) {
-          errs = errs.concat(await getValibotError(props.state, props.schema))
-        } else if (isSuperStructSchema(props.schema)) {
-          errs = errs.concat(await getSuperStructErrors(props.state, props.schema))
+        const schema = parseSchema(props.state)
+        const [errors, result] = await schema(props.schema as unknown as Schema)
+        if (errors) {
+          errs = errs.concat(errors)
         } else {
-          throw new Error('Form validation failed: Unsupported form schema')
+          form.value = result
```

Comment:
I think we can mutate `props.state` directly here with `Object.assign(props.state, result)` and remove the local ref.

Reply:
okee